smartsim/_core/mli/infrastructure/worker/worker.py

```diff
@@ -339,37 +366,49 @@ def has_raw_model(self) -> bool:
         """
         return self.raw_model is not None
 
-    @property
-    def raw_model(self) -> t.Optional[t.Any]:
-        """Returns the raw model to use to execute for this batch
-        if it is available.
-
-        :returns: A model if available, otherwise None"""
-        if self.has_valid_requests:
-            return self.requests[0].raw_model
-        return None
-
-    @property
-    def input_keys(self) -> t.List[TensorKey]:
-        """All input keys available in this batch's requests.
-
-        :returns: All input keys belonging to requests in this batch"""
-        keys = []
-        for request in self.requests:
-            keys.extend(request.input_keys)
-
-        return keys
-
-    @property
-    def output_keys(self) -> t.List[TensorKey]:
-        """All output keys available in this batch's requests.
-
-        :returns: All output keys belonging to requests in this batch"""
-        keys = []
-        for request in self.requests:
-            keys.extend(request.output_keys)
-
-        return keys
+    @classmethod
+    def from_requests(
+        cls,
+        requests: t.List[InferenceRequest],
+        inputs: t.Optional[TransformInputResult],
+        model_id: ModelIdentifier,
```

Comment:
Do we need `model_id`? Is it reliable? It makes some sense that we're sending a whole batch for processing by the same model...

But - if I put a request on the wrong queue, the code as-is would probably blow up because it doesn't confirm all the requests in the batch have the same model id as what was passed... If we say we trust the dispatcher, why send the model ID instead of just grabbing from the first request in the batch?

Reply:
Is this unnecessary?

```
    @classmethod
    def from_requests(
        cls,
        requests: t.List[InferenceRequest],
        model_id: ModelIdentifier,
    ) -> "RequestBatch":
        """Create a RequestBatch from a list of requests.
        ...
        """
        my_requests = [r for r in requests if r.model_key == model_id]
        not_my_requests = [r for r in requests if r.model_key != model_id]

        if not_my_requests:
            logger.warning("Attempted inference call with model ID mismatch")
```

If yes, why not do:

```
return cls(model_id=requests[0].model_key, ...
```

Reply:
including @al-rigazzi in case he knows this off the top of his head

Reply:
I'll wait for Al to chime in, but I totally see what you're saying!

Reply:
I just realized that a direct-inference model wouldn't have the model ID. I think I was wrong.. Removing model_id makes it not work, doesn't it?

Reply:
I think I'd prefer keeping it as a value passed in because `model_id=requests[0].model_key` could potentially be `None`? I'm not sure anymore lol